src/ppx/application.cpp

```diff
@@ -786,6 +786,15 @@ void Application::ShutdownMetrics()
     StopMetricsRun();
 }
 
+metrics::GaugeBasicStatistics Application::GetGaugeBasicStatistics(metrics::MetricID id) const
+{
+    if (!mStandardOpts.pEnableMetrics->GetValue()) {
```

Comment:
Perhaps this should produce a "stronger" assert/error, cause otherwise the data is invalid, and you might not realize. I guess depends on the usage though, whether it is only called if metrics are enabled, or can be called as a noop otherwise.

Reply:
changed into assert